lib/clustering/reclusterer.rb

```diff
@@ -31,8 +37,41 @@ def recluster
       end
     end
 
+    # The removed_ocns include an OCN no longer found in the remaining components
+    def ocns_changed?
+      return false if @removed_ocns.none?
+
+      (@removed_ocns - @cluster.component_ocns).any?
+    end
+
+    # Determines if all component OCNs are connected.
+    # Uses simple tests before attempting more involved graph traversal.
+    def needs_recluster?
+      return false if @cluster.ocns.one? ||
+        resolution_includes_cluster_ocns? ||
+        htitem_includes_all_cluster_ocns? ||
+        cluster_components.one?
```

Comment:
I'm not sure this one saves us much time - if there's only one thing in the cluster, then the graph traversal will be fast anyway; if there are more things, then it just means we're iterating over items in the cluster an extra time.